### PR TITLE
[SYCL] Replace OpenCL API with Native API in error reporting.

### PR DESCRIPTION
**Pull request by smaslov-intel**
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

**Review by vladimirlaz — DISMISSED** [review]

File: sycl/include/CL/sycl/detail/common.hpp [hunk 1]
```diff
@@ -100,10 +100,10 @@ static inline std::string codeToString(cl_int code) {
 #endif // #ifdef __SYCL_DEVICE_ONLY__
 
 #define __SYCL_OCL_ERROR_REPORT                                                \
```
Comment (bader) [inline sycl/include/CL/sycl/detail/common.hpp:102]: I think `__SYCL_OCL_ERROR_REPORT` macro should be updated as well.

`__SYCL_OCL_ERROR_REPORT` -> `__SYCL_NATIVE_ERROR_REPORT`
`__SYCL_SUPPRESS_OCL_ERROR_REPORT` -> `__SYCL_SUPPRESS_NATIVE_ERROR_REPORT`

Probably it make sense to keep support for `__SYCL_SUPPRESS_OCL_ERROR_REPORT` in addition for a while, but report a message that it's deprecated and is going to be removed.

Update: it applies to all defines in `common.hpp` file with `_OCL` in the name: `_OCL` -> `_NATIVE`.

Comment (smaslov-intel) [reply sycl/include/CL/sycl/detail/common.hpp:102]: Sure. There are so many internals that need to be renamed from OpenCL, and I want to just quickly fix what is visible to the user. Let's not have this gated by other renaming, which I currently don't have bandwidth to work on.

Comment (bader) [reply sycl/include/CL/sycl/detail/common.hpp:102]: Okay. Let's leave a TODO comment or GitHub issue to address this later.

File: sycl/include/CL/sycl/detail/common.hpp [hunk 2]
```diff
@@ -100,10 +100,10 @@ static inline std::string codeToString(cl_int code) {
...
-  "OpenCL API failed. " /*__FILE__*/                                           \
+  "Native API failed. " /*__FILE__*/                                           \
   /* TODO: replace __FILE__ to report only relative path*/                     \
   /* ":" __SYCL_STRINGIFY_LINE(__LINE__) ": " */                               \
-                               "OpenCL API returns: "
+                               "Native API returns: "
 
 #ifndef __SYCL_SUPPRESS_OCL_ERROR_REPORT
 #include <iostream>
```
Comment (smaslov-intel) [inline sycl/include/CL/sycl/detail/common.hpp:109]:
```suggestion
#include <iostream>
// TODO: rename all names with direct use of OCL/OPENCL to be backend agnostic.
```

File: sycl/include/CL/sycl/detail/common.hpp (hunk 1 above)
Comment (smaslov-intel) [reply sycl/include/CL/sycl/detail/common.hpp:102]: Added TODO comment.

**Review by bader — APPROVED** [review]

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**